tethysext/atcore/models/file_database/file_database_client.py

```diff
@@ -71,3 +77,78 @@ def path(self, root_dir: str) -> None:
         root_dir (str): the directory to be the root directory of the file database.
         """
         self._path = os.path.join(root_dir, str(self.instance.id))
+
+    def get_collection(self, collection_id: uuid.UUID) -> FileCollectionClient:
+        """
+        Get a FileCollectionClient owned by this FileDatabase by its collection_id
+
+        Args:
+            collection_id (uuid.UUID): The id for the file collection owned by this FileDatabase.
+
+        Returns:
+            The FileCollectionClient for the FileCollection.
+        """
+        collection_query = self._session.query(FileCollection).filter_by(
+            id=collection_id, file_database_id=self.instance.id
+        )
+        if collection_query.count() == 0:
+            raise FileCollectionNotFoundError(f'Collection with id "{str(collection_id)}" could not '
+                                              f'be found with this database.')
+        file_collection = collection_query.first()
+        collection_client = FileCollectionClient(
```

Comment:
This would be a case where it would be nice to just pass an instance instead of the ID. That way we won't have to do two queries for this operation. What do you think?

Reply:
It's up to you, but i think it is okay. The instance is lazy loaded on the other side. So it isn't an immediate 2nd query. It will only query again when you try to use the instance. That might be bad, but i don't think it is bad enough to change the design. I don't actually need to query the object out here if you know a better way just to check that it exists. I already have the id i need from the arguments. So if there is an efficient way to check if it exists that would probably be better.